.github/workflows/sonar_branch_analysis.yml

```diff
@@ -0,0 +1,94 @@
+---
+name: Sonar analysis for long-lived branches
+
+on:
+  push:
+    branches:
+      - main
+      - candlepin-*-HOTFIX
+
+env:
+  JAVA_DISTRIBUTION: 'temurin'
+  JAVA_VERSION: '17'
+
+jobs:
+  branch_unit_tests:
+    name: unit tests for long-lived branches
+    runs-on: ubuntu-latest
+    container:
+      image: fedora:38
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v3
+
+      - name: Install dependencies
+        shell: bash
+        run: dnf --setopt install_weak_deps=False install -y gettext jss
+
+      - name: Set up Java
+        uses: actions/setup-java@v3
+        with:
+          distribution: ${{ env.JAVA_DISTRIBUTION }}
+          java-version: ${{ env.JAVA_VERSION }}
+
+      - name: Run unit tests
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: test coverage
+
+      - name: Archive reports
+        uses: actions/upload-artifact@v3
+        with:
+          name: jacocoReports
+          path: |
+            **/build/test-results/**/*.xml
+            **/build/jacoco/test.exec
+
+  branch_sonar_analysis:
```

Comment:
This can be a single job. We don't really care about unit-test job results like we do in the PR workflow so separating the won't give us anything.

Reply:
I've consolidated this into a single job, which means we also don't need to archive/download the artifacts